configs/components/ruby-2.7.8.rb

```diff
@@ -203,19 +207,21 @@
   rbconfig_changes = {}
   if platform.is_aix?
     rbconfig_changes["CC"] = "gcc"
-  elsif platform.is_cross_compiled? || platform.is_solaris?
+  elsif platform.is_macos? && platform.architecture == 'arm64'
     if platform.name =~ /osx-11/
       rbconfig_changes["CC"] = 'clang -target arm64-apple-macos11'
     elsif platform.name =~ /osx-12/
       rbconfig_changes["CC"] = 'clang -target arm64-apple-macos12'
-    else
-      rbconfig_changes["CC"] = "gcc"
-      if platform.is_solaris?
-        # this should probably be done for aix and cross compiled targets too
-        rbconfig_changes["CPP"] = "gcc -E"
-      end
-      rbconfig_changes["warnflags"] = "-Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wno-long-long -Wno-missing-field-initializers -Wno-tautological-compare -Wno-parentheses-equality -Wno-constant-logical-operand -Wno-self-assign -Wunused-variable -Wimplicit-int -Wpointer-arith -Wwrite-strings -Wdeclaration-after-statement -Wimplicit-function-declaration -Wdeprecated-declarations -Wno-packed-bitfield-compat -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -Wno-maybe-uninitialized"
+    elsif platform.name =~ /osx-13/
+      rbconfig_changes["CC"] = 'clang -target arm64-apple-macos13'
```

Comment:
Similar comment here as in https://github.com/puppetlabs/puppet-runtime/pull/680/files#r1253820661. Since we're native compiling for macOS 13 ARM, we shouldn't need to specify the `target` and instead just specify the compiler, similar to the AIX case earlier on:

```suggestion
      rbconfig_changes["CC"] = 'clang'
```